include/swift/AST/AutoDiff.h

```diff
@@ -187,9 +187,13 @@ class AutoDiffParameterIndicesBuilder {
   llvm::SmallBitVector parameters;
 
 public:
-  /// Start building an `AutoDiffParameterIndices` for the given function type.
-  AutoDiffParameterIndicesBuilder(AnyFunctionType *functionType,
-                                  bool setAllParams = false);
+  /// Creates a `AutoDiffParameterIndicesBuilder` for the given function type.
+  AutoDiffParameterIndicesBuilder(AnyFunctionType *functionType);
+
+  /// Creates a `AutoDiffParameterIndicesBuilder` for the given function type,
+  /// inferring all differentiation parameters.
+  static AutoDiffParameterIndicesBuilder inferParameters(
```

Comment:
It would be useful to document the inference rules here.

Reply:
Inference rules documented in https://github.com/apple/swift/pull/23350/commits/487fc215654af3737f7dcaefd9c6c95224d9a554.